tests/Schema/TableDiffTest.php

```diff
@@ -29,6 +33,59 @@ public function testReturnsName(): void
         self::assertEquals(new Identifier('foo'), $tableDiff->getName($this->platform));
     }
 
+    public function testRenamedColumnDeprecationLayer(): void
+    {
+        $this->expectDeprecationWithIdentifier('https://github.com/doctrine/dbal/pull/6080');
+
+        /** @psalm-suppress InvalidArgument */
+        $diff = new TableDiff(
+            'foo',
+            [],
+            [
+                new ColumnDiff(
+                    'foo',
+                    new Column('foo', Type::getType(Types::INTEGER)),
+                    ['type'],
+                    new Column('foo', Type::getType(Types::BIGINT)),
+                ),
+                new ColumnDiff(
+                    'ba',
+                    new Column('ba', Type::getType(Types::INTEGER)),
+                    ['type'],
+                    new Column('ba', Type::getType(Types::BIGINT)),
+                ),
+            ],
+            [],
+            [],
+            [],
+            [],
+            new Table('foo'),
+            [],
+            [],
+            [],
+            [
+                'foo' => new Column('baz', Type::getType(Types::INTEGER)),
+                'bar' => new Column('renamed', Type::getType(Types::INTEGER)),
+            ],
+            [],
+        );
+
+        self::assertCount(3, $diff->getChangedColumns());
+        self::assertCount(2, $diff->getModifiedColumns());
+        self::assertEquals('foo', $diff->getChangedColumns()[0]->getOldColumnName()->getName());
+        self::assertEquals('baz', $diff->getChangedColumns()[0]->getNewColumn()->getName());
+        self::assertTrue($diff->getChangedColumns()[0]->hasTypeChanged());
+        self::assertEquals(Type::getType(Types::INTEGER), $diff->getChangedColumns()[0]->getNewColumn()->getType());
+        self::assertEquals('bar', $diff->getChangedColumns()[2]->getOldColumnName()->getName());
+        self::assertEquals('renamed', $diff->getChangedColumns()[2]->getNewColumn()->getName());
+
+        self::assertCount(2, $diff->renamedColumns);
+
+        $diff->renamedColumns = ['old_name' => new Column('new_name', Type::getType(Types::INTEGER))];
```

Comment:
@derrabus This new abstraction layer is not perfect and contains a small breaking change I think, it's not possible to do `$diff->renamedColumns['old_name'] = new Column(..)` (throws an exception), it was marked as internal though and in laravel it's used like in the test case https://github.com/laravel/framework/blob/10.x/src/Illuminate/Database/Schema/Grammars/RenameColumn.php#L64-L67

So I think it's an acceptable internal breaking change

Reply:
> So I think it's an acceptable internal breaking change

Totally.

> It is otherwise fixable but would involve the creation of a new Arrayable class to be returned from `__get` to call the setter when changes are pushed to the array

Let's not do down that road. 🙈